src/assisted_test_infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -599,6 +600,40 @@ def set_ipxe_url(self, network_name: str, ipxe_url: str):
         self.destroy_network(network)
         self.create_network(xml.toprettyxml())
 
+    def _create_dns_host_xml(self, api_vip: str, cluster_name: str):
+        """
+        Creates an XML object for adding a DNS host to a libvirt network
+        """
+        host_element = Element("host", ip=api_vip)
+        if cluster_name != "":
+            hostname = f"api.{cluster_name}.{self._entity_config.base_dns_domain}"
+            hostname_element = SubElement(host_element, "hostname")
+            hostname_element.text = hostname
+        return minidom.parseString(tostring(host_element))
+
+    def add_dns_host_to_network(self, network_name: str, api_vip: str, cluster_name: str):
+        """
+        Adds a dns-host to an existing network
+        """
+        host_xml = self._create_dns_host_xml(api_vip, cluster_name)
+        network = self.get_network_by_name(network_name)
+        network.update(
+            libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
+            libvirt.VIR_NETWORK_SECTION_DNS_HOST,
+            -1,
+            host_xml.toprettyxml(),
+        )
```

Comment:
This might be more readable
```suggestion
        network.update(
            libvirt.VIR_NETWORK_UPDATE_COMMAND_ADD_LAST,
            libvirt.VIR_NETWORK_SECTION_DNS_HOST,
            parentIndex=-1,
            xml=host_xml.toprettyxml(),
        )
```

Reply:
Modified, thank you!